_posts/2021-09-26-where-the-store-is-stored.md

```diff
@@ -0,0 +1,292 @@
+all too well (sorry colleagues!). I've actually already [written about
+it](/tech/posts/2020-11-22-state-store-in-frontend-codebases) in the past.
+
+I have changed companies a few month ago to join BlaBlaCar and the very cool
+frontend team building and maintaining its web app and find myself in situations
+that make me change how I think about the "frontend Store dilemma" I kept
+referring to. There is now a new question I ask myself again and again looking
+at our frontend codebases:
+
+Where is the store stored?
+
+# State vs Store - one more time 
+
+Let's go back to the difference between the concepts of *State* and *Store*
+even though we sometimes use the two terms indistinctively in frontend
+codebases. In every frontend app, data (the State) is stored somewhere (the
+Store).
+
+When we're thinking about states, we're in the world of ideas. We ask
+questions about what those states represent? Which transition between states are
+valid? How should we shape our states?
+
+When we deal with stores, we're not in the world of ideas anymore. We're back
+in the reality of the code that runs somewhere, mainly in a Javascript runtime.
+We ask where do store out data? How do we read it? How do we write it? How can
+we subscribe to changes?
+
+In this post, we're focusing on "store" kind of questions.
+
+# The multiple places we can store our stores
+
+The reality of the store that lives "somewhere" in the code is the very
+problem. The store in itself, whether a simple variable or a full Redux store,
+has be stored somewhere in our browser, NodeJS or other fancy javascript
+engine runtime. 
+
+Deciding *where our stores are stored* is not trivial and comes with big
+impacts. Let me draw here the three main potential locations we can use for our
+stores.
+
+## Modules
+
+We can use modules to host our stores. It's quite convenient and it makes full
+use of the Javascript ES6 language.
+
+Let's say we have a module storing Cars, our state here.
+
+```js
+// cars/store.js
+
+const _cars = []
+
+export function add(car) {
+  _cars.push(car);
+}
+
+export function list() {
+  return _cars;
+} 
+```
+
+This `cars/store.js` module acts as a very naive store with both *read*
+(`list`) and *write* (`add`) functionalities.
+
+Using this store in the rest of the codebase is very easy: we just have to use
+classic ES6 imports.
+
+```js
+// another/part/of/the/codebase.js
+import * as Cars from '../path/to/cars/store';
+
+function doingSomething() {
+  console.log('current cars: ', Cars.list());
+
+  const newCar = "My old car";
+
+  cars.add(newCar);
+}
+```
+
+And *voilà*! Because ES6 module are statefull, the `_cars` "private" variable
+will stay the same across all imports, making `cars/store.js` a store by itself.
+
+Should we need a reactive store (ie. being able to *subscribe* to state
+changes), we would write our own subscribe function by hand or use classic store
+librairies (some opinionated like Redux, Mobx, etc, some more transversal like
```

Comment:
I dont think the `etc` brings something here. Remove it, you already point out `some`, we understand it's not exhaustive. Here's my suggestion:

```
some opinionated like Redux, Mobx and some more transversal like the [@fp-51/store (https://github.com/fp51/store-library) I authored one or
two years ago
```